src/testdir/test_visual.vim

```diff
@@ -1630,4 +1630,40 @@ func Test_visual_substitute_visual()
   bwipe!
 endfunc
 
+func Test_visual_getregion()
+  new
+  call setline(1, ['one', 'two', 'three'])
+
+  call cursor(1, 1)
+  execute "normal!" "\<ESC>vl"
+  call assert_equal(['on'], getregion('v', '.'))
+  call assert_equal(['on'], getregion('.', 'v'))
+  call assert_equal(['o'], getregion('v', 'v'))
+  call assert_equal(['n'], getregion('.', '.'))
+
+  call cursor(1, 1)
+  execute "normal!" "\<ESC>Vll"
+  call assert_equal(['one'], getregion('v', '.'))
+  call assert_equal(['one'], getregion('.', 'v'))
+  call assert_equal(['one'], getregion('v', 'v'))
+  call assert_equal(['e'], getregion('.', '.'))
+
+  call cursor(1, 1)
+  execute "normal!" "\<ESC>\<C-v>ll"
+  call assert_equal(['one'], getregion('v', '.'))
+  call assert_equal(['one'], getregion('.', 'v'))
+  call assert_equal(['o'], getregion('v', 'v'))
+  call assert_equal(['e'], getregion('.', '.'))
+
+  call cursor(1, 1)
+  execute "normal!" "\<ESC>\<C-v>jj"
+  call assert_equal(['o', 't', 't'], getregion('v', '.'))
+
+  call cursor(1, 1)
+  execute "normal!" "\<ESC>\<C-v>jj$"
+  call assert_equal(['one', 'two', 'three'], getregion('v', '.'))
+
+  bwipe!
```

Comment:
Can you add a test for using this function in virtual mode and check for the additional space characters?

Reply:
What is the additional space characters?

Please provide the examples.

Reply:
The ones you add with visual block that extends past endofline

Reply:
I am referring to the following text in the help for getregion():

```
If |virtualmode| is enabled and selection is past the end of line, result lines are filled.
```

You can add the following test:

```
  " Virtual edit
  set virtualedit=all
  call cursor(1, 3)
  call feedkeys("\<Esc>\<C-v>lljj", 'xt')
  call assert_equal(['e  ', 'o  ', 'ree'], getregion('v', '.'))
  set virtualedit&
```